### PR TITLE
Upgrade to Lucene 9.8.0

### PR DESCRIPTION
**Pull request by tteofili**
Upgrade Anserini to use Apache Lucene 9.8.0.

### CODE REVIEW

**Comment by tteofili** [comment]
closes #679

File: src/main/java/io/anserini/ltr/DocumentFieldContext.java [hunk 1]
```diff
@@ -23,11 +23,7 @@
 import org.apache.lucene.index.Term;
 import org.apache.lucene.index.Terms;
 import org.apache.lucene.index.TermsEnum;
-import org.apache.lucene.search.DocIdSetIterator;
-import org.apache.lucene.search.DocValuesFieldExistsQuery;
-import org.apache.lucene.search.IndexSearcher;
-import org.apache.lucene.search.Query;
-import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.*;
```
Comment (lintool) [inline src/main/java/io/anserini/ltr/DocumentFieldContext.java:26]: Let's change back to explicit imports? I like that better stylistically...

**Review by lintool — APPROVED** [review]

**Comment by lintool** [comment]
Let's go ahead and merge! (And I'll run all our regressions from scratch anyway...)

File: src/main/java/io/anserini/util/ExtractDocumentLengths.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public static void main(String[] args) throws Exception {
 
     out.println("internal_docid\texternal_docid\tdoc_length\tunique_term_count\tlossy_doc_length\tlossy_unique_term_count");
     for (int i = 0; i < numDocs; i++) {
-      Terms terms = reader.getTermVector(i, Constants.CONTENTS);
+      Terms terms = reader.termVectors().get(i, Constants.CONTENTS);
```
Comment (jpountz) [inline src/main/java/io/anserini/util/ExtractDocumentLengths.java:81]: Ideally you'd fetch the `reader.termVectors()` object only once, before the loop, and then reuse it in the loop. (And similarly to other places that rely on stored fields and term vectors.)

Comment (tteofili) [reply src/main/java/io/anserini/util/ExtractDocumentLengths.java:81]: yeah, good point!

Comment (tteofili) [reply src/main/java/io/anserini/util/ExtractDocumentLengths.java:81]: Will have another PR fix that in different places (and perhaps  storedFields() too)

**Pull request merged**